subsys/bluetooth/host/gatt.c

```diff
@@ -180,13 +183,127 @@ BT_GATT_SERVICE_DEFINE(_2_gap_svc,
 #endif
 );
 
-#if defined(CONFIG_BT_GATT_DYNAMIC_DB)
-static void sc_ccc_cfg_changed(const struct bt_gatt_attr *attr,
-			       u16_t value)
+#if defined(CONFIG_BT_GATT_SERVICE_CHANGED)
+struct sc_data {
+	u16_t start;
+	u16_t end;
+} __packed;
+
+struct gatt_sc_cfg {
+	u8_t		id;
+	bt_addr_le_t	peer;
+	struct sc_data	data;
+};
+
+#define SC_CFG_MAX (CONFIG_BT_MAX_PAIRED + CONFIG_BT_MAX_CONN)
+static struct gatt_sc_cfg sc_cfg[SC_CFG_MAX] = {};
+
+static struct gatt_sc_cfg *find_sc_cfg(u8_t id, bt_addr_le_t *addr)
+{
+	BT_DBG("id: %u, addr: %s", id, bt_addr_le_str(addr));
+
+	for (size_t i = 0; i < ARRAY_SIZE(sc_cfg); i++) {
+		if (id == sc_cfg[i].id &&
+		    !bt_addr_le_cmp(&sc_cfg[i].peer, addr)) {
+			return &sc_cfg[i];
+		}
+	}
+
+	return NULL;
+}
+
+#if defined(CONFIG_BT_SETTINGS)
```

Comment:
Is this #if really needed? What you you drop it and use IS_ENABLED in the places where you need to call this function?

Reply:
sc_store(...) uses bt_settings_encode_key(...) and settings_save_one(...), which are only compiled in when CONFIG_BT_SETTINGS and CONFIG_SETTINGS are set.

If you prefer I could however put the #if inside the function (as in sc_clear) and use IS_ENABLED when calling it.

Reply:
What you describe sounds exactly like the use case that IS_ENABLED exists for. As long as the code can see the prototypes of the functions in question in their header files it doesn't matter if they're part of the final image or not, since the calls to them don't exist anymore at the linking stage (the compiler will have stripped them out)

Reply:
Yes that's right, will test this right away! Sorry I'm not used to those IS_ENABLED thingies.

Reply:
Alright, latest push addresses this and makes more use of IS_ENABLED now that I know the tricks. I tested that it builds without Warnings with multiple config scenarios and it seems to work just fine and not affect the image size at all!